pkg/proxy/iptables/proxier.go

```diff
@@ -571,45 +616,61 @@ func (proxier *Proxier) OnEndpointsSynced() {
 	proxier.syncProxyRules(syncReasonEndpoints)
 }
 
-// Convert a slice of api.Endpoints objects into a map of service-port -> endpoints.
-func buildNewEndpointsMap(allEndpoints endpointsMap, curMap proxyEndpointMap, hostname string) (newMap proxyEndpointMap, hcEndpoints map[types.NamespacedName]int, staleSet map[endpointServicePair]bool) {
-
-	// return values
-	newMap = make(proxyEndpointMap)
-	hcEndpoints = make(map[types.NamespacedName]int)
+func updateEndpointsMap(
+	endpointsMap proxyEndpointsMap,
```

Comment:
This is my least favorite facet of Go.  One of, anyway.

Please comment that this is is modified in this function.  Same with the staleSet arg to detectStaleConnections